lxd/images.go

```diff
@@ -1989,9 +1989,44 @@ func pruneExpiredImagesTask(d *Daemon) (task.Func, task.Schedule) {
 
 func pruneLeftoverImages(d *Daemon) {
 	opRun := func(op *operations.Operation) error {
+		// Check if dealing with shared image storage.
+		var storageImages string
+		err := d.State().Node.Transaction(func(tx *db.NodeTx) error {
+			nodeConfig, err := node.ConfigLoad(tx)
+			if err != nil {
+				return err
+			}
+
+			storageImages = nodeConfig.StorageImagesVolume()
+
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+
+		if storageImages != "" {
+			// Parse the source.
+			fields := strings.Split(storageImages, "/")
+			if len(fields) != 2 {
+				return fmt.Errorf("Invalid syntax for volume, must be <pool>/<volume>")
+			}
+
+			// Load the pool.
+			pool, err := storagePools.GetPoolByName(d.State(), fields[0])
+			if err != nil {
+				return err
+			}
+
+			// Skip cleanup if image volume may be multi-node.
```

Comment:
This comment would be good if it was expanded to include the *why* too, so from the PR description something like:

"As some storage pool drivers support shared image storage, avoid deleting images downloaded by other servers".